WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -248,13 +304,31 @@ final class ProductImageUploader: ProductImageUploaderProtocol {
     }
 
     private func scheduleUploadInProgressNotificationIfNeeded() {
-        guard !activeUploadsPublisher.isEmpty else { return }
+        if featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) {
+            userDefaultsStatuses.statusesPublisher
+                .map { statuses in
+                    statuses.filter { $0.isUploading }
+                }
+                .sink { uploadingStatuses in
+                    if !uploadingStatuses.isEmpty {
+                        let notification = LocalNotification(scenario: .productImageBackgroundUpload)
+                        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+                        Task {
+                            await LocalNotificationScheduler(pushNotesManager: ServiceLocator.pushNotesManager).schedule(notification: notification,
+                                                                                                                         trigger: trigger, remoteFeatureFlag: nil)
+                        }
+                    }
+                }
+                .store(in: &cancellables)
```

Comment:
I think observing the storage is not necessary here as this method is called in the foreground before the app is sent to the background. 

You should be able to access the statuses directly through the storage's `getAllStatuses`.

Reply:
Makes sense and good catch. I updated it here: 2607997ee9e1ca6c59d598300d22e50e9dd63c23